modules/tensor_mechanics/include/materials/RankTwoCartesianComponent.h

```diff
@@ -0,0 +1,40 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Material.h"
+#include "RankTwoTensor.h"
+
+/// ADRankTwoCartesianComponent computes selected components from a Rank-2 tensors.
+class RankTwoCartesianComponent : public Material
```

Comment:
Ok, `RankTwoCartesianComponent` and `ADRankTwoCartesianComponent` should really just be typedefs of the same `RankTwoCartesianComponentTempl<bool is_ad>` class template to avoid the code duplication.

Reply:
I agree.  I did it this way so that the PR I will create tomorrow to resolve the GenericMaterialProperty can be resolved and use the typedefs

Reply:
Yes. it should. This is a temporary workaround until we can fix the issue with `getGenericMaterialProperty` giving us the wrong qp index. As soon as we merge this PR, we are planning to submit another PR that combines them using the templates, and will fail so that someone can help us fix that. I don't want to hold this capability up for that.